File: 2026POTD/JANUARY/04-01-2026/1390.go
func sumFourDivisors(nums []int) int {
	ans := 0

	for _, x := range nums {
		n := x
		p, q, cnt := 0, 0, 0

		for i := 2; i*i <= n && cnt <= 2; i++ {
			if n%i == 0 {
				cnt++
				if cnt == 1 {
					p = i
				} else {
					q = i
				}
				for n%i == 0 {
					n /= i
				}
			}
		}

		if n > 1 {
			cnt++
			if cnt == 1 {
				p = n
			} else {
				q = n
			}
		}

		if cnt == 2 && p*q == x {
			ans += 1 + p + q + x
		} else if cnt == 1 && p*p*p == x {
			ans += 1 + p + p*p + x
		}
	}

	return ans
}

File: 2026POTD/JANUARY/04-01-2026/1390.rb
# @param {Integer[]} nums
# @return {Integer}
def sum_four_divisors(nums)
  ans = 0

  nums.each do |x|
    n = x
    p = q = cnt = 0

    i = 2
    while i * i <= n && cnt <= 2
      if n % i == 0
        cnt += 1
        cnt == 1 ? p = i : q = i
        n /= i while n % i == 0
      end
      i += 1
    end

    if n > 1
      cnt += 1
      cnt == 1 ? p = n : q = n
    end

    if cnt == 2 && p * q == x
      ans += 1 + p + q + x
    elsif cnt == 1 && p * p * p == x
      ans += 1 + p + p * p + x
    end
  end

  ans
end